lib/spack/spack/operating_systems/_base_classes.py

```diff
@@ -0,0 +1,36 @@
+# Copyright 2013-2021 Lawrence Livermore National Security, LLC and other
```

Comment:
Why is this called `_base_classes`, it contains only one class. Might as well just call it `operating_system`?

Reply:
Code navigation is a lot easier if the file name corresponds with the class name... I always struggle to find AutotoolsPackage etc too because they are in build_systems/autotools.py, which is confusing.

Reply:
Renaming done. I named those modules `_base_classes`:
1. To make clear where the base classes for the package were implemented
2. To be able to import the module for the base class first, given the import ordering rules we adopt

That said, I'm not particularly attached to that naming and if people find it more readable otherwise I'm absolutely fine with it.

Reply:
Done in 0ebda4d753f6e7f7570c98ae2314a63d959237c6